pkg/chains/signing.go

```diff
@@ -222,11 +235,31 @@ func (o *ObjectSigner) Sign(ctx context.Context, tektonObj objects.TektonObject)
 	// Now mark the TektonObject as signed
 	if err := MarkSigned(ctx, tektonObj, o.Pipelineclientset, extraAnnotations); err != nil {
 		return err
+	} else {
+		logger.Infof("Recording count for stored payload for %v", tektonObj)
+		measureMetrics("mrcount", o.Recorder, ctx)
 	}
-
 	return nil
 }
 
+func measureMetrics(metrictype string, mtr MetricsRecorder, ctx context.Context) {
+	logger := logging.FromContext(ctx)
+	logger.Infof("Type of Metrics Recorder ", mtr, metrictype)
+	if mtr != nil {
+		switch mt := metrictype; mt {
+		case "sgcount":
+			mtr.RecordCountMetrics("sgcount", ctx)
```

Comment:
This just adds complexity. Instead of switch and case in two places, Add four different functions.

Reply:
Or you can try to use Generics if possible.

Reply:
Reduced switch statement to 1